src/BenchmarkDotNet.Diagnostics.Windows/EtwProfilerConfig.cs

```diff
@@ -10,33 +12,38 @@ namespace BenchmarkDotNet.Diagnostics.Windows
     public class EtwProfilerConfig
     {
         public const ulong MatchAnyKeywords = ulong.MaxValue;
-        
+
         public bool PerformExtraBenchmarksRun { get; }
-        
+
         public int BufferSizeInMb { get; }
-        
+
         public float CpuSampleIntervalInMiliseconds { get; }
 
         public KernelTraceEventParser.Keywords KernelKeywords { get; }
 
         public IReadOnlyDictionary<HardwareCounter, Func<ProfileSourceInfo, int>> IntervalSelectors { get; }
-        
+
         public IReadOnlyCollection<(Guid providerGuid, TraceEventLevel providerLevel, ulong keywords, TraceEventProviderOptions options)> Providers { get; }
 
+        public bool CreateHeapSession { get; }
+
         /// <param name="performExtraBenchmarksRun">if set to true, benchmarks will be executed one more time with the profiler attached. If set to false, there will be no extra run but the results will contain overhead. True by default.</param>
         /// <param name="bufferSizeInMb">ETW session buffer size, in MB. 256 by default</param>
         /// <param name="cpuSampleIntervalInMiliseconds">The rate at which CPU samples are collected. By default this is 1 (once a millisecond per CPU). There is a lower bound on this (typically 0.125 ms)</param>
         /// <param name="intervalSelectors">interval per hardware counter, if not provided then default values will be used.</param>
         /// <param name="kernelKeywords">kernel session keywords, ImageLoad (for native stack frames) and Profile (for CPU Stacks) are the defaults</param>
         /// <param name="providers">providers that should be enabled, if not provided then default values will be used</param>
+        /// <param name="createHeapSession">value indicating whether to create heap session</param>
```

Comment:
nit: I would add few more words to make it more clear to the end users that most probably they don't need to set it to true.

```suggestion
        /// <param name="createHeapSession">value indicating whether to create heap session. False by default, used internally by NativeMemoryDiagnoser.</param>
```